File: transactionApi/models/Transaction.cs
namespace TransactionApi.Models;

    public class Transaction{

    public int Id {get;set;}
    public  required int AccountId {get;set;}
    public required decimal Amount {get;set;}
    public DateTime TransactionDate {get;set;} = DateTime.UtcNow;
    public required  int TransactionTypeId { get; set; }
    public  TransactionType? TransactionType { get; set; }

    }

File: banking-ui/models/transaction.ts


export  interface Transaction {
    id:number;
    customerName: string;
    accountNo: number;
    destinationAccountNo?: number;
    transactionType: TransactionType;
    amount: number;
    transactionDate: Date;
    
    
}

export interface TransactionType{
    id:number;
    name: string;
    isInterBank: boolean;
    requiresDestinationAccount: boolean;
    
    
}

export interface TransactionInitiationDto {

    accountNo: number;
    destinationAccountNo?: number;
    amount: number;
    transactionTypeId: number;
    
    
    
}